src/atomic_file/atomic.rs

```diff
@@ -116,14 +117,41 @@ impl AtomicFile {
         Ok(Self { directory, prefix })
     }
 
-    fn latest_version(&self) -> Result<usize> {
-        let mut max_version = 0;
-        for entry in fs::read_dir(&self.directory)? {
-            if let Some(version) = parse_version(&entry?.file_name()) {
-                max_version = std::cmp::max(max_version, version);
-            }
-        }
-        Ok(max_version)
+    /// Return a vec of files with latest version and the latest version. Multiples files can be found if they comes from different sources.
+    /// For example one from cellphone and one from computer can both a a version 2.
+    fn latest_version(&self) -> Result<(Vec<ReadOnlyFile>, usize)> {
+        let files_iterator = fs::read_dir(&self.directory)?.flatten();
+        let (files, version) = files_iterator.into_iter().fold(
+            (vec![], 0),
+            |(mut files, mut max_version), entry| {
```

Comment:
Let's call it `curr_max_version` or something like this. Can be confused with `MAX_VERSION_FILES` constant.